src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -501,28 +393,112 @@ private static MavenArtifactType GetMavenArtifactType(string fileName)
                 // Split the version content into its individual parts.
                 // [0] - The version
                 // [1] - The date it was published
-                // [2] - The time it waas published
-                // [3] - The download count
+                // [2] - The time it was published
+                // [3] - The file size in bytes
                 string[] versionParts = versionContent.Split(new[] { ' ' }, StringSplitOptions.RemoveEmptyEntries);
                 if (DateTime.TryParse($"{versionParts[1]} {versionParts[2]}", out DateTime publishDateTime))
                 {
                     return publishDateTime;
                 }
             }
-            else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
+
+            // If the directory index approach does not work, try to get the "Last-Modified" header from the .pom file.
+            baseUrl = $"{feedUrl.EnsureTrailingSlash()}{packageNamespace}/{packageName}/{packageVersion}/{packageName}-{packageVersion}.pom";
+            HttpResponseMessage response = await httpClient.GetAsync(baseUrl);
+
+            if (response.Content.Headers.TryGetValues("Last-Modified", out var values))
+            {
+                var lastModified = DateTime.Parse(values.First());
+                return lastModified;
+            }
+
+            return null;
+        }
+
+        private async IAsyncEnumerable<string> GetArtifactDownloadUris_DirectoryIndexStrategyAsync(string baseUrl, bool useCache = true)
+        {
+            HttpClient httpClient = CreateHttpClient();
+            var fileNameList = new List<string>();
+
+            try
+            {
+                string? html = await GetHttpStringCache(httpClient, baseUrl, useCache);
+                if (string.IsNullOrEmpty(html))
+                {
+                    throw new InvalidOperationException();
```

Comment:
How could we make this easier to understand/debug if we end up throwing this exception?

Also - if we throw an exception, that means that we will not attempt any other of the strategies for finding directory contents. Is that intentional?

Reply:
Same note applies to the other strategies as well.

Reply:
There's a catch block, so the other strategies should still be attempted. I can add some logging in the catch block so we can capture info on this exception.